lib/LibraryTemplatePlugin.js

```diff
@@ -34,47 +42,61 @@ LibraryTemplatePlugin.prototype.apply = function(compiler) {
 	compiler.plugin("this-compilation", function(compilation) {
 		switch(this.target) {
 			case "var":
-				compilation.apply(new SetVarMainTemplatePlugin("var " + accessorAccess(false, this.name)));
+				var varExpressions = this.names.map(function(name) {
+					return "var " + accessorAccess(false, name);
+				});
+				compilation.apply(new SetVarMainTemplatePlugin(varExpressions));
 				break;
 			case "assign":
-				compilation.apply(new SetVarMainTemplatePlugin(accessorAccess(undefined, this.name)));
+				var varExpressions = this.names.map(function(name) {
+					return accessorAccess(undefined, name);
+				});
+				compilation.apply(new SetVarMainTemplatePlugin(varExpressions));
 				break;
 			case "this":
 			case "window":
 			case "global":
-				if(this.name)
-					compilation.apply(new SetVarMainTemplatePlugin(accessorAccess(this.target, this.name)));
+				if(this.names) {
+					var varExpressions = this.names.map(function(name) {
+						return accessorAccess(this.target, name);
+					});
+					compilation.apply(new SetVarMainTemplatePlugin(varExpressions));
+				}
 				else
-					compilation.apply(new SetVarMainTemplatePlugin(this.target, true));
+					compilation.apply(new SetVarMainTemplatePlugin([this.target], true));
```

Comment:
I changed it here for the minor performance benefit as if it's not an array here the extra code will run in `SetVarMainTemplatePlugin` to convert it. Happy to revert though